library/src/c_api.rs

```diff
@@ -119,7 +115,7 @@ pub extern "C" fn shorebird_init(
 pub extern "C" fn shorebird_next_boot_patch_number() -> *mut c_char {
     log_on_error(
         || {
-            let patch = updater::next_boot_patch().context("failed to fetch patch info")?;
+            let patch = updater::next_boot_patch()?.context("failed to fetch patch info")?;
```

Comment:
Is this now a Result<Result<T>>?  .context is a method on result (from anyhow) to add context to the potential error.

Reply:
I think this is because it's now Result<Option>?

Reply:
This calls ? to unwrap the Result<Option<T>> into Option<T> which then context() turns back into a Result<T> which we then unwrap again. 🤣   We should just be able to return the option w/o logging?